### PR TITLE
Update Settings design: Add feature flag & new Settings screen

### PR DESCRIPTION
**Pull request by mikescamell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1208785611228225/f

### Description

Adds a feature flag and new screen for the new updated Settings screen designs. 

For now it's the same layout as before, this is just laying the groundwork for the rest of the implementation. 

So you can see that the flag is working, the new Settings screen has a temporary title of "New Settings" which will be removed when the UI changes enough that it's perceivable.

I also added the ability to trigger notifications from dev settings so it's easier to test the ClearData notification route to opening settings.

### Steps to test this PR

- [x] Turn on the feature flag "newSettings" 

_Opening Settings from the Browser menu_
- [x] Click overflow menu
- [x] Press Settings button
- [x] Ensure "New Settings" is displayed at the top

_Opening Settings from ADS shortcut_
- [x] Add the ADS shortcut via phone homescreen
- [x] Press Shortcut 
- [x] ADS screen should open
- [x] Press back
- [x] Ensure "New Settings" is displayed at the top

_Clear Data Notification_
- [x] Open new Notifications screen in Dev Settings
- [x] Click "Data clearing set to manual" 
- [x] Click on the notification
- [x] Ensure "New Settings" is displayed at the top

_VPN Disabled Notification_
- [x] Open new Notifications screen in Dev Settings
- [x] Click "NetP Disabled" 
- [x] Click on the notification
- [x] Ensure "New Settings" is displayed at the top

_Clear Data Notification_
- [x] Open new Notifications screen in Dev Settings
- [x] Click "Data clearing set to manual" 
- [x] Click on the notification
- [x] Ensure "New Settings" is displayed at the top

_New Tab Page_
- [x] Ensure "pluginNewTabPage" feature flag is enabled
- [x] Open a new tab
- [x] Click the "Settings" shortcut
- [x] Ensure "New Settings" is displayed at the top

_Tab Switcher_
- [x] Open Tab Switcher screen
- [x] Click overflow menu
- [x] Click "Settings"
- [x] Ensure "New Settings" is displayed at the top

_Privacy Pro_
- [x] Simplest way to see it is to edit ln 126 in `SpecialUrlDetector` to true and load a web page 
- [x] Press back
- [x] New settings should be visible

### UI changes

N/A

### CODE REVIEW

**Comment by mikescamell** [comment]
* **#5300** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5300?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5300?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by malmstein — CHANGES_REQUESTED** [review]
I don’t understand why have you copied the screen. What’s the benefit of doing it? We should start the new screen from scratch.

There’s a feature flag, have it disabled by default also for internal users. When you want to try the new screen, manually flip the switch and test it.

File: app/src/main/java/com/duckduckgo/app/settings/NewSettingsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+@SuppressLint("NoLifecycleObserver")
+@ContributesViewModel(ActivityScope::class)
+class NewSettingsViewModel @Inject constructor(
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/settings/NewSettingsViewModel.kt:73]: same here, this is inheriting all the bad practices we’ve been adding over the years. It’s going to take you much longer than if you start the screen from scratch with the new design.

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/RealSubscriptions.kt [hunk 2]
```diff
@@ -175,7 +175,10 @@ class PrivacyProFeatureStore @Inject constructor(
         }
     }
 
-    private fun SharedPreferences.save(key: String, state: State) {
+    private fun SharedPreferences.save(
```
Comment (malmstein) [inline subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/RealSubscriptions.kt:178]: is this needed?

Comment (mikescamell) [reply subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/RealSubscriptions.kt:178]: Just a formatting issue.

File: settings/settings-api/src/main/java/com/duckduckgo/settings/api/NewSettingsFeature.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package com.duckduckgo.settings.api
+
+import com.duckduckgo.feature.toggles.api.Toggle
+
+interface NewSettingsFeature {
+
+    @Toggle.DefaultValue(true)
```
Comment (malmstein) [inline settings/settings-api/src/main/java/com/duckduckgo/settings/api/NewSettingsFeature.kt:23]: this is enabled by default to EVERYONE. it should be false

**Review by malmstein — CHANGES_REQUESTED** [review]
The toggle should be disabled by default

**Review by malmstein — APPROVED** [review]
works as expected, thanks @mikescamell !

**Pull request merged**